src/Microsoft.AspNet.Localization/RequestLocalizationMiddleware.cs

```diff
@@ -142,6 +156,39 @@ private CultureInfo GetCultureInfo(IList<string> cultures, IList<CultureInfo> su
                 }
             }
 
+            if (fallbackToAncestorCulture & currentDepth < MaxCultureFallbackDepth)
+            {
+                // Walk backwards through the culture list and remove any root cultures (those with no parent)
+                for (var i = cultureNames.Count - 1; i >= 0; i--)
+                {
+                    var cultureName = cultureNames[i];
+                    if (cultureName != null)
+                    {
+                        var lastIndexOfHyphen = cultureName.LastIndexOf('-');
+                        if (lastIndexOfHyphen > 0)
```

Comment:
@rynowak no, leading and trailing hyphens are not valid, more naming rules at MSDN [CultureAndRegionInfoBuilder](https://msdn.microsoft.com/en-us/library/system.globalization.cultureandregioninfobuilder.cultureandregioninfobuilder.aspx) (which does not mean you cannot get it from a client of course)

@DamianEdwards  For the record, there is a fundamental difference between parsing and `CultureInfo.Parent`. There are parent chains like:
`zh-Hans-HK -> zh-CHS -> zh-Hans -> zh -> Invariant` or 
`zh-TW -> zh-CHT -> zh-Hant -> zh -> Invariant` or 
`mn-MN -> mn-Cyrl -> mn -> Invariant` etc.
Supporting simplified and traditional Chinese while expecting all of `zh-CN`, `zh-HK`, `zh-MO`, `zh-SG`, `zh-TW`, `zh-CHS` and `zh-CHT` to resolve to the correct one sounds reasonable (and common) to me. I am also sure there was a political review for these decisions. Anyway, someone from the team should probably decide about these costs vs performance.